diff --git a/nodejs/crypto.mjs b/nodejs/crypto.mjs
index 314f7fb2fdf084a98b7dd6d07fed8259467cf411..3e138a0538b82a731b8ca1e449314615e0a47fef 100644
--- a/nodejs/crypto.mjs
+++ b/nodejs/crypto.mjs
@@ -2,7 +2,7 @@
 // global scope. We import the generated JS file to make `wasm_bindgen`
 // available which we need to initialize our WASM code.
 import { parentPort } from 'node:worker_threads';
-import Comlink from 'comlink';
+import * as Comlink from 'comlink';
 import nodeEndpoint from 'comlink/dist/esm/node-adapter.mjs';
 import { CryptoUtils } from './crypto-wasm/index.js';
 
diff --git a/nodejs/index.mjs b/nodejs/index.mjs
index eae02e62c3b1268e377114ec88c967b48a75b2af..2d94b9a716f98ab50597c142f17752f00e80272f 100644
--- a/nodejs/index.mjs
+++ b/nodejs/index.mjs
@@ -1,5 +1,5 @@
 import { Worker } from 'node:worker_threads';
-import Comlink from 'comlink';
+import * as Comlink from 'comlink';
 import nodeEndpoint from 'comlink/dist/esm/node-adapter.min.mjs';
 import { Address, CryptoUtils, Transaction } from './main-wasm/index.js';
 import { clientFactory } from '../launcher/node/client-proxy.mjs';
diff --git a/nodejs/worker.mjs b/nodejs/worker.mjs
index e78edd6a31cc0bd495b956e27fe50b81495a3c38..2e9a633ae788782c66872e3730cdfd5daf0eabd1 100644
--- a/nodejs/worker.mjs
+++ b/nodejs/worker.mjs
@@ -2,7 +2,7 @@
 // global scope. We import the generated JS file to make `wasm_bindgen`
 // available which we need to initialize our WASM code.
 import { parentPort } from 'node:worker_threads';
-import Comlink from 'comlink';
+import * as Comlink from 'comlink';
 import nodeEndpoint from 'comlink/dist/esm/node-adapter.mjs';
 import websocket from 'websocket';
 import { Client } from './worker-wasm/index.js';
diff --git a/package.json b/package.json
index d3195f0f28c986e5130fccc441c3aaf77586ff4f..c398637f2db21feca0e7144e506254db4237f7b2 100644
--- a/package.json
+++ b/package.json
@@ -24,6 +24,7 @@
     },
     "./web": {
       "browser": "./web/index.js",
+      "import": "./web/index.js",
       "types": "./types/web.d.ts"
     }
   },
diff --git a/web/index.js b/web/index.js
index fc7904583a00fc3b3438cb370be03b37ee1078d7..d2d0124a0e3b52dd698ec0c257206c049e7a0378 100644
--- a/web/index.js
+++ b/web/index.js
@@ -1,29 +1,4 @@
-import * as Comlink from './comlink.min.mjs';
-import init, { Address, CryptoUtils, Transaction } from './main-wasm/index.js';
-import { clientFactory } from '../launcher/browser/client-proxy.mjs';
-import { cryptoUtilsWorkerFactory } from '../launcher/browser/cryptoutils-worker-proxy.mjs';
-import { setupMainThreadTransferHandlers } from '../launcher/browser/transfer-handlers.mjs';
-
-setupMainThreadTransferHandlers(Comlink, {
-    Address,
-    Transaction,
-});
-
-const Client = clientFactory(
-    () => new Worker(new URL('./worker.js', import.meta.url)),
-    worker => Comlink.wrap(worker),
-);
-
-const CryptoUtilsWorker = cryptoUtilsWorkerFactory(
-    () => new Worker(new URL('./crypto.js', import.meta.url)),
-    worker => Comlink.wrap(worker),
-);
-for (const propName in CryptoUtilsWorker) {
-    const prop = CryptoUtilsWorker[propName];
-    if (typeof prop === 'function') {
-        CryptoUtils[propName] = prop;
-    }
-}
+import init from './main-wasm/index.js';
 
 export * from './main-wasm/index.js';
 export { Client };
